README.md

```diff
@@ -1,6 +1,130 @@
 # application-aware-quota
 Empower users to customize and manage resource quotas per component in Kubernetes, utilizing the scheduling gates feature.
 
+## Motivation
+In a multi-tenant cluster where multiple users or teams share a Kubernetes cluster with a fixed number of nodes, 
+there is a risk that one team might consume more than its fair share of resources. 
+Resource quotas are a tool for administrators to manage and mitigate this concern.
+
+However, the native ResourceQuotas in Kubernetes are admission-based, which can cause the resource quota 
+admission controller to become a bottleneck for the API Server during evaluation and ResourceQuota status update calls. 
+Additionally, native ResourceQuotas are designed to handle predefined pod resources such as `requests.cpu`, 
+`limits.cpu`, `requests.memory`, `limits.memory`, and more. 
+This design is not easily extendable due to the potential burden on the API server, exacerbating the bottleneck.
+
+Native pod resources are often considered implementation details that can change across different development 
+stages or versions of Kubernetes Operators. For example, consider KubeVirt:
+
+KubeVirt is an operator that enables the deployment and management of virtual machines alongside containerized applications. 
+KubeVirt workloads run inside pods, presenting specific challenges with resource quotas:
+
+1. VM Live Migration: During VM live migration, both source and target pods are required.
+Having both pods simultaneously affects quotas, potentially leading to migration issues and upgrade failures.
+2. Billing Considerations: Service providers typically bill VMs similarly to platforms 
+like OpenStack, based on RAM size and CPU threads. However, KubeVirt's pods must account 
+for infrastructure overhead, which complicates resource quota management.
+
+
+As a result, deploying KubeVirt in a multi-tenant cluster is highly challenging. 
+It requires significant resource allocation to handle live migrations and manage VM infrastructure 
+overhead. This also couples users to the VM implementation, making them dependent on infrastructure 
+overhead that can change unexpectedly.
+
+
+## Overcoming Resource Quota Challenges with ApplicationAwareResourceQuota
+
+To address these challenges, the AAQ Operator use a different approach compared to the 
+native ResourceQuota. To avoid API server overhead, the AAQ Operator introduces an alternative 
+quota implementation called ApplicationAwareResourceQuota.
+The ApplicationAwareResourceQuota is API compatible with the native ResourceQuota, meaning it shares the same [specification and status definitions](https://github.com/kubevirt/application-aware-quota-api/blob/ac6757c09496bfab205d499a41daa6ed8815eab6/pkg/apis/core/v1alpha1/types.go#L46).
```

Comment:
Bear in mind that the code link references a certain commit.
Perhaps referencing main is better?
https://github.com/kubevirt/application-aware-quota-api/blob/main/pkg/apis/core/v1alpha1/types.go#L46

You can also use a permalink: https://github.com/kubevirt/application-aware-quota-api/blob/ac6757c09496bfab205d499a41daa6ed8815eab6/pkg/apis/core/v1alpha1/types.go#L46.

The pro is that the code will stay the same, so if the code updates it will always refer to the same code. This is also a con since on code updates the docs should be updated. WDYT?